### PR TITLE
Make it possible to eval() all floating-point values dumped by string()

### PR DESCRIPTION
**Pull request by justinmk**

### CODE REVIEW

**Comment by ZyX-I** [comment]
I suggested to disable `-Werror` only in quickbuild, not in all CI services. I also do not like current solution, `-Wc11-extensions` should be disabled for quickbuild only.

**Comment by justinmk** [comment]
> Wc11-extensions should be disabled for quickbuild/freebsd only.

I agree but I don't know how to do that.

**Comment by ZyX-I** [comment]
@justinmk Newest commit in my PR should work.

File: src/nvim/eval.c [hunk 1]
```diff
@@ -4001,12 +4001,22 @@ eval6 (
...
+          f1 = (f2 == 0
+                ? (
+#ifdef NAN
+                    f1 == 0
+                    ? NAN
+                    :
+#endif
+                    (f1 > 0
+                     ? INFINITY
+                     : -INFINITY)
```
Comment (oni-link) [inline src/nvim/eval.c]: If quiet NaNs are not supported `-INFINITY` is returned for `0.0/0.0`. Should this be mentioned in the docs?

**Pull request closed without merging**